src/mame/drivers/8080bw.cpp

```diff
@@ -3560,6 +3560,423 @@ void _8080bw_state::init_invmulti()
+	Adolfo Melilli (adolfo@melilli.com).
+	Those disks where retrieved by Alessandro Bolgia (xadhoom76@gmail.com) and Lorenzo Fongaro (lorenzo.fongaro@virgilio.it) and 
+	dumped by Piero Andreini (pieroandreini@gmail.com) using KryoFlux hardware and software.
+	Subsequently Jean Paul Piccato (j2pguard-spam@yahoo.com) mounted the images and compiled the source files, managed to set up a
+	romset and wrote a mame driver that aims to reproduce in the most faithful way the work of Melilli at Model Racing in late '70s.
+ 
+	The game driver is not based on hardware inspection and is solely derived from assumptions I've made looking at the assembler
+	code and comments written into the source files of the game. Several of those hypothesis came following the directions of 
+	previous yet contemporary Model Racing works (Eg: Claybuster) and where confirmed by Melilli himself.
+ 
+	Being unreleased this games has not an official name, thus the name used in the source files was used instead.
+
+***********************************************************************************************************************************/
+
+u8 orbite_state::orbite_scattered_colorram_r(ATTR_UNUSED address_space &space, ATTR_UNUSED offs_t offset, ATTR_UNUSED u8 mem_mask)
+{
+	return m_scattered_colorram[(offset & 0x1f) | ((offset & 0x1f80) >> 2)];
+}
+
+
+void orbite_state::orbite_scattered_colorram_w(ATTR_UNUSED address_space &space, ATTR_UNUSED offs_t offset, ATTR_UNUSED u8 data, ATTR_UNUSED u8 mem_mask)
+{
+	m_scattered_colorram[(offset & 0x1f) | ((offset & 0x1f80) >> 2)] = data;
+}
+
+
+void orbite_state::orbite_map(address_map &map)
+{
+//  map(0x0000, 0x1fff).rom();
+	map(0x0000, 0x07ff).rom();
+	map(0x0800, 0x1fff).rom();
+	map(0x2000, 0x3fff).ram().share("main_ram");
+	map(0xc000, 0xdfff).rw(FUNC(orbite_state::orbite_scattered_colorram_r), FUNC(orbite_state::orbite_scattered_colorram_w));
+}
+
+
+void orbite_state::orbite_io_map(address_map &map)
+{
+	map(0x06, 0x06).w(m_watchdog, FUNC(watchdog_timer_device::reset_w));
+
+	// Ports verified on sources
+	map(0x08, 0x08).r(m_mb14241, FUNC(mb14241_device::shift_result_r));
+	map(0x20, 0x20).w(m_mb14241, FUNC(mb14241_device::shift_count_w));
+	map(0x40, 0x40).w(m_mb14241, FUNC(mb14241_device::shift_data_w));
+
+	map(0x66, 0x66).portr("IN0");
+	map(0x76, 0x76).portr("IN1");
+	map(0x7A, 0x7A).portr("IN2");
+}
+
+
+static INPUT_PORTS_START( orbite )
+	PORT_START("IN0")
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_JOYSTICK_RIGHT ) PORT_2WAY
+	PORT_BIT( 0x80, IP_ACTIVE_HIGH, IPT_UNUSED )
+
+	PORT_START("IN1")
+	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_JOYSTICK_LEFT ) PORT_2WAY
+
+	PORT_START("IN2")   // port 2
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_BUTTON1 )
+INPUT_PORTS_END
+
+
+void orbite_state::machine_start()
+{
+	m_scattered_colorram = std::make_unique<uint8_t []>(0x800);  
+	save_pointer(&m_scattered_colorram[0], "m_scattered_colorram", 0x800);
+	mw8080bw_state::machine_start();  
```

Comment:
In general it's better to call the base `machine_start` first.

Reply:
Thank you for your support. Sadly I've a lot to learn on mame internal working.

Reply:
Don't worry - MAME can seem big and unapproachable, but you can work your way up from small tasks to bigger tasks.  We're trying to make the core easier to work on and more approachable, too.